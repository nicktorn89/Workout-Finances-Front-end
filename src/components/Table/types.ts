import { WorkoutObject } from 'src/store/modules/types';

export interface TableProps {
  data: WorkoutObject[];
  onEdit: (id: string) => void;
  onCheckboxChange: (e: React.MouseEvent<HTMLElement, MouseEvent>) => void;
}
