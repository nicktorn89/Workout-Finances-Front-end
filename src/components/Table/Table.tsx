import React, { memo } from 'react';
import moment from 'moment';

import { TableProps } from './types';

import TableBody from '@material-ui/core/TableBody';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Checkbox from '@material-ui/core/Checkbox';
import Done from '@material-ui/icons/Done';
import Edit from '@material-ui/icons/Edit';

import { MaterialTable, MaterialPaper, TableHeadCell, TableBodyCell } from './styled';
import { createData } from 'src/modules/Main/utils';

const Table: React.FC<TableProps> = ({ data, onCheckboxChange, onEdit }) => {
  const rows = data.map((row) => createData(row));

  const handleEdit = (e: React.MouseEvent<SVGSVGElement, MouseEvent>): void => {
    onEdit(e.currentTarget.dataset.dbId as string);
  };

  return (
    <MaterialPaper>
      <MaterialTable>
        <TableHead>
          <TableRow>
            <TableHeadCell />
            <TableHeadCell>Изменить</TableHeadCell>
            <TableHeadCell>Дата</TableHeadCell>
            <TableHeadCell>Кол-во человек</TableHeadCell>
            <TableHeadCell>Прибыль</TableHeadCell>
            <TableHeadCell>Бесплатная</TableHeadCell>
            <TableHeadCell>Персональная</TableHeadCell>
            <TableHeadCell>Джампы</TableHeadCell>
          </TableRow>
        </TableHead>

        <TableBody>
          {rows.map((row, index) => {
            const rowMonth = moment(row.date).month() + 1 < 10
              ? `0${moment(row.date).month()}`
              : moment(row.date).month() + 1;

            const rowHour = moment(row.date).hour() < 10
              ? `0${moment(row.date).hour()}`
              : moment(row.date).hour() + 1;

            const rowMinute = moment(row.date).minute() < 10
              ? `0${moment(row.date).minute()}`
              : moment(row.date).minute();

            return (
              <TableRow key={index}>
                <TableBodyCell>
                  <Checkbox
                    name={`${index}`}
                    onClick={onCheckboxChange}
                  />
                </TableBodyCell>
                <TableBodyCell>
                  <Edit onClick={handleEdit} data-db-id={row.dataId} />
                </TableBodyCell>
                <TableBodyCell
                  component='th'
                  className='date-column'
                  scope='row'
                  align='left'
                >
                  {`${moment(row.date).date()}.${rowMonth}.${moment(row.date).year()} 
                ${rowHour}:${rowMinute}`}
                </TableBodyCell>
                <TableBodyCell
                  align='left'
                >
                  {row.peopleCount}
                </TableBodyCell>
                <TableBodyCell
                  align='left'
                >
                  {row.salary}
                </TableBodyCell>
              </TableRow>
            );
          })}
        </TableBody>
      </MaterialTable>
    </MaterialPaper>
  );
};

export default memo(Table);
