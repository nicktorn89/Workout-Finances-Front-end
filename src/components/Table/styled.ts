import styled from 'styled-components';
import Table from '@material-ui/core/Table';
import Paper from '@material-ui/core/Paper';
import TableCell from '@material-ui/core/TableCell';
import { media } from 'src/theme/mixin';

// @ts-nocheck
export const TableHeadCell: any = styled(TableCell)`
  &&& {
    font-size: 1.1rem;
    font-weight: bold;
  }
`;

export const TableBodyCell: any = styled(TableCell)`
  &&& {
    font-size: 1rem;
  }
`;

export const MaterialTable: any = styled(Table)`
  &&& {
    .date-column {
      text-align: center;
      line-height: 1.5;
    }

    width: 80%;
  
    ${media.phone`
      width: 90%;
      overflow-x: auto;
      display: block;
    `}
  
    ${media.desktop`
      display: table;
    `}
  }
`;

export const MaterialPaper: any = styled(Paper)`
  margin: 2em auto 3rem;

  ${media.phone`
    width: 90%;
  `}

  ${media.desktop`
    width: 80%;
  `}
  
  ${media.tablet`
    width: 80%;
  `}
`;
