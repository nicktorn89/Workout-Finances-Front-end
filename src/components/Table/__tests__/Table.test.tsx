import React, { MouseEvent as ReactMouseEvent } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Table from '../Table';
import empty from 'ramda/es/empty';

const addIndexes = (e: ReactMouseEvent<HTMLElement, MouseEvent>) => {
};

describe('<Slider />', () => {
  const TableComponent = shallow(<Table onEdit={empty} onCheckboxChange={addIndexes} data={[]} />);
  const TableComponentToJson = shallowToJson(TableComponent);
  
  it('Should be equal to snapshot', () => {
    expect(TableComponentToJson).toMatchSnapshot();
  });
});
