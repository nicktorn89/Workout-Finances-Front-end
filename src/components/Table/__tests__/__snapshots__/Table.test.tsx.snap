// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Should be equal to snapshot 1`] = `
<Styled(WithStyles(Paper))>
  <Styled(WithStyles(Table))>
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <Styled(WithStyles(TableCell)) />
        <Styled(WithStyles(TableCell))>
          Изменить
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Дата
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Кол-во человек
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Прибыль
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Бесплатная
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Персональная
        </Styled(WithStyles(TableCell))>
        <Styled(WithStyles(TableCell))>
          Джампы
        </Styled(WithStyles(TableCell))>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody) />
  </Styled(WithStyles(Table))>
</Styled(WithStyles(Paper))>
`;
